BRANCHES.md

```diff
@@ -2,24 +2,33 @@
 
 At any point in time, we have a number of maintained branches consisting of:
 
-- the development branch: this is where new features lands, as well as bug
-  fixes and security fixes
-- one or more LTS branches: these only get bug fixes and security fixes.
+- The [`master`](https://github.com/ARMmbed/mbedtls/tree/master) branch:
+  this always contains the latest release, including all publicly available
+  security fixes.
+- The [`development`](https://github.com/ARMmbed/mbedtls/tree/development) branch:
+  this is where new features land,
+  as well as bug fixes and security fixes.
+- One or more long-time support (LTS) branches:
+  these only get bug fixes and security fixes.
 
 We use [Semantic Versioning](https://semver.org/). In particular, we maintain
-API compatibility in the development branch between major version changes. We
+API compatibility in the `master` branch between major version changes. We
```

Comment:
I would make the usual D&I comment here that I thought we were moving away from using 'master', but I guess this can change when we make that change